app/javascript/components/student/TracksList.jsx

```diff
@@ -46,16 +47,15 @@ export function TracksList({ statusOptions, tagOptions, ...props }) {
         </div>
       </div>
       <div className="md-container">
-        {isSuccess && <Header data={data} query={request.query} />}
+        <Header latestData={latestData} query={request.query} />
```

Comment:
Ah, so this is the trick where we don't conditionally render the element but instead use the latest data. Nice!

Are there more places where we should be doing this?

Reply:
I think we will need to adapt this pattern more and more in the future. But let's leave that for later I think.